File: src/main/java/org/mifos/fineract/services/SelfSpmApi.java
package org.mifos.fineract.services;

import org.mifos.fineract.models.SurveyData;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import rx.Observable;

import java.util.List;

public interface SelfSpmApi {
    /**
     * @return Observable&lt;List&lt;SurveyData&gt;&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("self/surveys")
    Observable<List<SurveyData>> fetchAllSurveys();


}

File: src/main/kotlin/org/openapitools/client/apis/SelfSpmApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.SurveyData
import retrofit2.http.GET

interface SelfSpmApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.collections.List<SurveyData>]
     */
    @GET("v1/self/surveys")
    suspend fun fetchAllSurveys(): List<SurveyData>

}
